plugins/workflow/notification/notification.php

```diff
@@ -0,0 +1,252 @@
+			return;
+		}
+
+		return $this->enhanceTransitionForm($form, $data);
+	}
+
+	/**
+	 * Add different parameter options to the transition view, we need when executing the transition
+	 *
+	 * @param   Form      $form  The form
+	 * @param   stdClass  $data  The data
+	 *
+	 * @return  boolean
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	protected function enhanceTransitionForm(Form $form, $data)
+	{
+		Form::addFormPath(__DIR__ . '/forms');
+
+		$form->loadFile('workflow_notification');
+
+		return true;
+	}
+
+	/**
+	 * Send a Notification to defined users a transion is performed
+	 *
+	 * @param   string   $context  The context for the content passed to the plugin.
+	 * @param   array    $pks      A list of primary key ids of the content that has changed stage.
+	 * @param   object  $data    Object containing data about the transition
+	 *
+	 * @return  boolean
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function onWorkflowAfterTransition($context, $pks, $data)
+	{
+		// Check if send-mail is active
+		if (empty($data->options['send_mail']) || !$data->options['send_mail'])
+		{
+			return true;
+		}
+
+		// ID of the items whose state has changed. 
+		$pks = ArrayHelper::toInteger($pks);
+
+		if (empty($pks))
+		{
+			return true;
+		}
+
+		// Get UserIds of Receivers
+		$userIds = $this->getUsersFromGroup($data);
+
+		// If there are no receivers, stop here
+		if (empty($userIds))
+		{
+			return true;
+		}
+
+		// Prepare Language for messages
+		$default_language = ComponentHelper::getParams('com_languages')->get('administrator');
+		$debug = $this->app->get('debug_lang');
+
+		// Get the Model of the Item via $context
+		$parts = explode('.', $context);
+
+		$component = $this->app->bootComponent($parts[0]);
+
+		$modelName = $component->getModelName($context);
+
+		$model = $component->getMVCFactory()->createModel($modelName, $this->app->getName(),  ['ignore_request' => true]);
+
+		// Add author of the item to the receivers arry if the param email-author is set
+		if (!empty($data->options['email_author']) && !empty($item->created_by))
+		{
+			$author = $this->app->getIdentity($item->created_by);
+
+			if (!empty($author) && !$author->block)
+			{
+				if (!in_array($author->id, $userIds))
+				{
+					$userIds[] = $author->id;
+				}
+			}
+		}
+
+		// Get the model for private messages
+		$model_message = $this->app->bootComponent('com_messages')
+					->getMVCFactory()->createModel('Message', 'Administrator');
+
+		// Get the title of the stage
+		$model_stage = $this->app->bootComponent('com_workflow')
+					->getMVCFactory()->createModel('Stage', 'Administrator');
+
+		$toStage = $model_stage->getItem($data->to_stage_id)->title;
+
+		// The active user
+		$user = $this->app->getIdentity();
+
+		foreach ($pks as $pk)
+		{
+			// Get the item whose state has been changed
+			$item = $model->getItem($pk);
+
+			// Send Email to receivers
+			foreach ($userIds as $user_id)
+			{
+				$receiver = User::getInstance($user_id);
```

Comment:
Perhaps there should be a check, so we don't send to ourself?
https://github.com/bembelimen/joomla-cms/pull/92/files#diff-4bec224e62bf4c1a53784d49773e3841L686